### ISSUE TITLE
Move grammar to the official grammars repository

### ISSUE THREAD

**Issue by KvanTTT** [tshprecher/antlr_psql#9]
Hi! What do you think about adding your grammar to official grammars repository [grammars-v4](https://github.com/antlr/grammars-v4)? It would be easier to maintain and fix by the community. There is at least one request on it: https://github.com/antlr/grammars-v4/issues/1501

**Comment by mebelousov** [comment]
Hi, @KvanTTT !
Please check [test_coverage.html](https://github.com/tshprecher/antlr_psql/blob/master/test_coverage.html). There are a lof of statements that parser cannot recognize.
As I found each parser for different SQL dialect in grammars-v4 has own way. 
Also I think that it is easier to maintain the standalone repository.

**Comment by tshprecher** [comment]
I agree eventually this can be in the official antlr/grammars-v4 directory, but I'm not sure it's stable enough yet. For example, I have an unmerged branch tackling some `ALTER` statements, and from time to time I add to it and will eventually merge with master.

When it gets to a stable state, I'll be fine with moving it to antlr/grammars-v4.

**Comment by KvanTTT** [comment]
> Please check test_coverage.html. There are a lof of statements that parser cannot recognize.

This is not a big deal. A lot of grammars in the repository have a lack of support for some syntax.

> As I found each parser for different SQL dialect in grammars-v4 has own way.

In general, yes, but they have a lot of things in common.

**Comment by tshprecher** [comment]
I'll tell you what. Let me scramble to get as much done of this by the end of Oct 1 (one month). There are things that bug me about the TODOs littered everywhere and the fact that there are tens of thousands of small individual test files that should be consolidated once this is more stable to avoid ballooning the size of antlr/grammars-v4.

It's just a little rough around the edges, but it's getting there. I do want to finish this, but life has gotten in the way.

**Comment by KvanTTT** [comment]
Ok, sure, take your time. No rush anyway.

**Comment by tshprecher** [comment]
I won't be able to finish this by Oct 1, but with the help @mebelousov we are making progress and I am committed to getting this done. I'm also coming around to @mebelousov's opinion above that this is easier to maintain as a standalone repository, especially as later versions of Postgres are also supported.

**Comment by mebelousov** [comment]
I hope that we could be done with the parser by year's end.

@tshprecher how about add some tags to the repository? Now antlr_psql is not included in search with "antlr4 postgresql" https://github.com/search?q=antlr4+postgresql

**Comment by tshprecher** [comment]
Good idea, @mebelousov. I added the tags and now the repo shows up in that search. I think we can get this done by the end of the year. The major queries (SELECT, INSERT, UPDATE, and DELETE) have mostly been tackled.
